sdk/python/kfp/compiler/_component_builder.py

```diff
@@ -58,6 +59,83 @@ def download_gcs_blob(local_path, gcs_path):
     blob = bucket.blob(gcs_blob)
     blob.download_to_filename(local_path)
 
+class VersionedDependency(object):
+  """ DependencyVersion specifies the versions """
+  def __init__(self, name, version=None, min_version=None, max_version=None):
+    """ if version is specified, no need for min_version or max_version;
+     if both are specified, version is adopted """
+    self._name = name
+    if version is not None:
+      self._min_version = version
+      self._max_version = version
+    else:
+      self._min_version = min_version
+      self._max_version = max_version
+
+  @property
+  def name(self):
+    return self._name
+
+  @property
```

Comment:
Why do we need a property here as opposed to a simple field?

Reply:
I understand this as I come from a C# background myself and like static typed languages. Python is different though so a change between a field and a read-write property is not a breaking change. In Python 3.6 we get @dataclass which makes things much easier:

```
@dataclass
class VersionedDependency:
  name: str
  min_version: Optional[str]
  max_version: Optional[str]
```

Reply:
Good to know about the @dataclass.